test/distributed.in.cpp

```diff
@@ -29,32 +29,67 @@
 using namespace std;
 using namespace ngraph;
 
-TEST(distributed_${BACKEND_NAME}, allreduce)
-{
+void test_allreduce_common_${BACKEND_NAME}(Reduce_t reduce_type) {
```

Comment:
There should be a `static` qualifier on this.

I don't think it's necessary to encode `${BACKEND_NAME}` in the function name, because `distributed.in.cpp` will be instantiated separately for each backend.